cpp/cmake/bob.cmake

```diff
@@ -154,6 +154,80 @@ function(bob_get_semver)
   message(STATUS "${PROJECT_NAME}_SEMVER = ${SEMVER}")
 endfunction(bob_get_semver)
 
+function(bob_cxx_pedantic_flags)
```

Comment:
I want to preserve the `bob_` prefix:
1. to avoid naming colision with CMake projects that will use this project.
1. to preserve consistency in bob functions.
1. so it remains easy to know where it comes from

Reply:
I sort of wish that CMake had namespaces…